learning_observer/learning_observer/auth/social_sso.py

```diff
@@ -211,7 +241,37 @@ async def _process_student_documents(student):
     # TODO saved skipped doc ids somewhere?
 
 
-async def _google(request):
+async def _handle_canvas_authorization(request):
+    '''
+    Handle Canvas authorization
+    '''
+    if 'error' in request.query:
+        return {}
+
+    token_uri = settings.pmss_settings.token_uri(types=['lms', 'canvas_oauth'])
+    url = token_uri
+
+    params = {
+        "grant_type": "refresh_token",
+        'client_id': settings.pmss_settings.client_id(types=['lms', 'canvas_oauth']),
+        'client_secret': settings.pmss_settings.client_secret(types=['lms', 'canvas_oauth']),
+        "refresh_token": settings.pmss_settings.refresh_token(types=['lms', 'canvas_oauth'])
```

Comment:
From my understanding of the Oauth workflow, the `refresh_token` should be tied to the user signing in and not generically tied to the system. This token is used to fetch new `access_tokens` when needed without requiring another sign-on.

The Google API code in LO did not implement the `refresh_token` functionality, though it should do something with it.